docs/core/tools/dotnet-nuget-trust.md

```diff
@@ -0,0 +1,319 @@
+
+  Specific NuGet file that should be used instead of the standard hierarchy.
+
+### `certificate`
+
+Adds a trusted signer with the given name, based on a certificate fingerprint.
+
+#### Synopsis
+
+```dotnetcli
+dotnet nuget trust certificate <name> <fingerprint> [OPTIONS]
+```
+
+#### Arguments
+
+- **`name`**
+
+  The name of existing trusted signer going be added. If a trusted signer with the given name already exists, the certificate item will be added to that signer. Otherwise a trusted author will be created with a certificate item from given certificate information.
+
+- **`<fingerprint>`**
+
+  The fingerprint of the certificate.
+
+#### OPTIONS:
+
+- **`--allow-untrusted-root`**
+
+  Specifies if the certificate for the trusted signer should be allowed to chain to an untrusted root. This is not recommended.
+
+- **`--algorithm <ALGORITHM>`**
+
+  Specifies the hash algorithm used to calculate the certificate fingerprint. Defaults to SHA256. Values supported are SHA256, SHA384 and SHA512.
+  
+- **`-h|--help`**
+
+  Prints out a description of how to use the command.
+
+- **`-v, --verbosity <LEVEL>`**
+
+  Set the verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+
+- **`--configfile <PATH>`**
+
+  Specific NuGet file that should be used instead of the standard hierarchy.
+
+### `source`
+
+Adds a trusted signer based on a given package source.
+
+#### Synopsis
+
+```dotnetcli
+dotnet nuget trust source <name> [source-url] [--owners <List>] [OPTIONS]
+```
+
+#### Arguments
+
+- **`name`**
+
+  The name of existing trusted signer going be added. If only `<name>` is provided without `<source-url>`, the package source from your NuGet configuration files with the same name will be added to the trusted list. If `<name>` already exists in the configuration, the package source will be appended to it.
+
+#### OPTIONS:
+
+- **`--owners <List>`**
+
+  Semi-colon separated list of trusted owners to further restrict the trust of a repository.
+
+- **`source-url`**
+
+  If a `source-url` is provided, it MUST be a v3 package source URL (like <https://api.nuget.org/v3/index.json>). Other package source types are not supported.
+
+- **`-h|--help`**
+
+  Prints out a description of how to use the command.
+
+- **`-v, --verbosity <LEVEL>`**
+
+  Set the verbosity level. Allowed values are q[uiet], m[inimal], n[ormal], d[etailed], and diag[nostic].
+
+- **`--configfile`**
+
+  Specific NuGet file that should be used instead of the standard hierarchy.
+
+## Examples
+
+- List trusted signers:
+
+  ```dotnetcli
+  dotnet nuget trust list
+  ```
+
+- Trust source *NuGet* in specified *nuget.config* file:
+
+  ```dotnetcli
+  dotnet nuget trust source NuGet --configfile ..\nuget.config
+  ```
+
+- Trust an author from signed nupkg package file *foo.nupkg* with *--allow-untrusted-root* option:
+
+  ```dotnetcli
+  dotnet nuget trust author PackageAuthor .\foo.nupkg --allow-untrusted-root
+  ```
```

Comment:
Given that we say this option is not recommended, should we omit an example showing it?